### ISSUE TITLE
error switching between fullscreen and annotator panel modes

### ISSUE THREAD

**Issue by childers** [GMOD/Apollo#1214]
I was just testing 2.0.4 and can log in, view it in fullscreen mode and go back and forth between fullscreen and Annotator Panel mode.  

When I go into an organisms browser by passing the name into the URL (good feature!), clicking on "Show Annotator Panel" results in a 404 error though I am not getting any messages in any of the tomcat logs.

Looking at the 404 URL, it looks like the new URL is missing the webapp name (in this case "apollo").  Adding the webapp name makes the URL work, though it is still resolving to the wrong scaffold.  Reseting the scaffold name manually causes the custom track to show up.  I'm including the URLs before and after clicking the Show Annotator Panel button.

It also looks like the after URL has some malformed strings in the highlight section, but I'm not sure why that would be populated at all.

Before:
```shell
https://apollo-dev.nal.usda.gov/apollo/Ephemera%20danica/jbrowse/?loc=Scaffold690%3A452898..453618&addStores=%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F3bb1517bfebf4046a89341cfcfef6afd%2FEdan07162013.scaffolds_new_ids.fa.gff%22%7D%7D&addTracks=%5B%7B%22label%22%3A%22BLAST%2B%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&tracks=Annotations&highlight=
```
After:
```shell
https://apollo-dev.nal.usda.gov/annotator/index?clientToken=92221766933785422662&addTracks=[{%22label%22:%22BLAST+%20Results%22,%22category%22:%220.%20Reference%20Assembly%22,%22type%22:%22WebApollo/View/Track/DraggableBLASTFeatures%22,%22store%22:%22url%22,%22style%22:{%22renderClassName%22:%22gray-center-10pct%22,%22subfeatureClasses%22:{%22match_part%22:%22blast-match_part%22}}}]&addStores={%22url%22:{%22type%22:%22JBrowse/Store/SeqFeature/GFF3%22,%22urlTemplate%22:%22https://gmod-dev.nal.usda.gov/webapp/media/blast/task/3bb1517bfebf4046a89341cfcfef6afd/Edan07162013.scaffolds_new_ids.fa.gff%22}}&tracks=Annotations&highlight=&nal.usda.gov/apollo/Ephemera%20danica/jbrowse/?loc=Scaffold690:452898..453618&nal=[usda.gov/apollo/Ephemera%20danica/jbrowse/?loc:Scaffold690:452898..453618,%20usda:[gov/apollo/Ephemera%20danica/jbrowse/?loc:Scaffold690:452898..453618]]
```

**Comment by childers** [comment]
oh and before I forget:
```shell

Apollo Genome Annotator

    Version: 2.0.4
    Grails version: 2.5.5
    Groovy version: 2.4.4
    JVM version: 1.7.0_111


```

**Comment by nathandunn** [comment]
@childers  I'm unable to reproduce it:

http://icebox.lbl.gov/Apollo2/Cow/jbrowse/index.html?loc=AY526085.1%3A6537..9804&tracks=DNA%2CAnnotations%2CRefSeq_Cow&highlight=

When I click on the annotator panel I go here:

http://icebox.lbl.gov/Apollo2/auth/login?targetUri=/annotator/loadLink%3Floc%3DAY526085.1%3A6537..9804%26tracks%3DDNA%2CAnnotations%2CRefSeq_Cow%26highlight%3D%26organism%3DCow%26clientToken%3D78233117414601593181480757533%26controller%3Dannotator%26format%3Dnull%26action%3DloadLink

If you clear the cache do you get the same issue (on either)?

**Comment by childers** [comment]
Were you logged in when you did clicked the link?   Using a private browser session, (and not logged in) clicking the show Annotator Panel button redirected to the login button, and upon logging in, I get the same 404 error.

It looks like the target URI in the second link is pointing to "/annotator", and not "/apollo/annotator/". If I understand right, would that redirect to the wrong place? The final URL still has the same strange reformatting in the end.  Could that be something funny happening when it is parsing the addStore and addTrack information?

I have a serious love/hate relationship with caching.

Initial:
```shell
https://apollo-dev.nal.usda.gov/apollo/auth/login?targetUri=/annotator/loadLink%3FaddTracks%3D%5B%7B%22label%22%3A%22BLAST%2B+Results%22%2C%22category%22%3A%220.+Reference+Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D%26addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F3bb1517bfebf4046a89341cfcfef6afd%2FEdan07162013.scaffolds_new_ids.fa.gff%22%7D%7D%26tracks%3DAnnotations%2CBLAST%2B+Results%26clientToken%3D924741302335978829392%26highlight%3D%26organism%3DEphemera+danica%26nal.usda.gov%2Fapollo%2FEphemera+danica%2Fjbrowse%2F%3Floc%3DScaffold548%3A135866..136490%26nal%3D%5Busda.gov%2Fapollo%2FEphemera+danica%2Fjbrowse%2F%3Floc%3AScaffold548%3A135866..136490%2C+usda%3A%5Bgov%2Fapollo%2FEphemera+danica%2Fjbrowse%2F%3Floc%3AScaffold548%3A135866..136490%5D%5D%26action%3DloadLink%26format%3Dnull%26controller%3Dannotator
```
After clicking Show Annotator PAnel:
```shell
https://apollo-dev.nal.usda.gov/apollo/auth/login?targetUri=/annotator/loadLink%3FaddTracks%3D%5B%7B%22label%22%3A%22BLAST%2B+Results%22%2C%22category%22%3A%220.+Reference+Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D%26addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F3bb1517bfebf4046a89341cfcfef6afd%2FEdan07162013.scaffolds_new_ids.fa.gff%22%7D%7D%26tracks%3DAnnotations%2CBLAST%2B+Results%26clientToken%3D924741302335978829392%26highlight%3D%26organism%3DEphemera+danica%26nal.usda.gov%2Fapollo%2FEphemera+danica%2Fjbrowse%2F%3Floc%3DScaffold548%3A135866..136490%26nal%3D%5Busda.gov%2Fapollo%2FEphemera+danica%2Fjbrowse%2F%3Floc%3AScaffold548%3A135866..136490%2C+usda%3A%5Bgov%2Fapollo%2FEphemera+danica%2Fjbrowse%2F%3Floc%3AScaffold548%3A135866..136490%5D%5D%26action%3DloadLink%26format%3Dnull%26controller%3Dannotator
```
After entering credentials and clicking Login:
```shell
https://apollo-dev.nal.usda.gov/annotator/loadLink?addTracks=[{%22label%22:%22BLAST+%20Results%22,%22category%22:%220.%20Reference%20Assembly%22,%22type%22:%22WebApollo/View/Track/DraggableBLASTFeatures%22,%22store%22:%22url%22,%22style%22:{%22renderClassName%22:%22gray-center-10pct%22,%22subfeatureClasses%22:{%22match_part%22:%22blast-match_part%22}}}]&addStores={%22url%22:{%22type%22:%22JBrowse/Store/SeqFeature/GFF3%22,%22urlTemplate%22:%22https://gmod-dev.nal.usda.gov/webapp/media/blast/task/3bb1517bfebf4046a89341cfcfef6afd/Edan07162013.scaffolds_new_ids.fa.gff%22}}&tracks=Annotations,BLAST+%20Results&clientToken=924741302335978829392&highlight=&organism=Ephemera%20danica&nal.usda.gov/apollo/Ephemera%20danica/jbrowse/?loc=Scaffold548:135866..136490&nal=[usda.gov/apollo/Ephemera%20danica/jbrowse/?loc:Scaffold548:135866..136490,%20usda:[gov/apollo/Ephemera%20danica/jbrowse/?loc:Scaffold548:135866..136490]]&action=loadLink&format=null&controller=annotator
```

**Comment by childers** [comment]
You can append the last part of this URL to simulate the string:
```shell
https://apollo.nal.usda.gov/baccuc/jbrowse/?loc=NW_011863694.1%3A841433..842249&addStores=%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fi5k.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F94eabfeca7ca4f8c8ff36cb0faa8fc80%2F28588_ref_asm80634v1_chrun_refseq_ids.fa.gff%22%7D%7D&addTracks=%5B%7B%22label%22%3A%22BLAST%2B%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&tracks=DNA%2CAnnotations%2CBLAST%2B%20Results&highlight=
```

Specifically:
```shell
&addStores=%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fi5k.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F94eabfeca7ca4f8c8ff36cb0faa8fc80%2F28588_ref_asm80634v1_chrun_refseq_ids.fa.gff%22%7D%7D&addTracks=%5B%7B%22label%22%3A%22BLAST%2B%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&tracks=DNA%2CAnnotations%2CBLAST%2B%20Results&highlight=
```

Would that be helpful for you?  The file shouldn't work, but we just care hoe the software processes the URL, right?

**Comment by nathandunn** [comment]
@childers  I will give it a shot.

**Comment by nathandunn** [comment]
@childers  I'm seeing the same behavior.  I'll take a look.   [This is the offending code](https://github.com/GMOD/Apollo/blob/master/grails-app/controllers/org/bbop/apollo/AnnotatorController.groovy#L48-L118), which should always use the same redirect URI, though not sure why this would work in some circumstances and not others.

**Comment by nathandunn** [comment]
I take that back the [funny code is here](https://github.com/GMOD/Apollo/blob/master/grails-app/controllers/org/bbop/apollo/AuthController.groovy#L54), which produces this code: 

```
2016-08-19 10:43:29,580 [http-bio-8080-exec-25] INFO  apollo.AuthController  - Redirecting to '/annotator/loadLink?addTracks=[{"label":"BLAST+ Results","category":"0. Reference Assembly","type":"WebApollo/View/Track/DraggableBLASTFeatures","store":"url","style":{"renderClassName":"gray-center-10pct","subfeatureClasses":{"match_part":"blast-match_part"}}}]&addStores={"url":{"type":"JBrowse/Store/SeqFeature/GFF3","urlTemplate":"https://i5k.nal.usda.gov/webapp/media/blast/task/94eabfeca7ca4f8c8ff36cb0faa8fc80/28588_ref_asm80634v1_chrun_refseq_ids.fa.gff"}}&loc=Group1.1:553102..829582&tracks=DNA,Annotations,Official Gene Set v3.2&clientToken=71738124710084550111357913116&highlight=&organism=4&action=loadLink&format=null&controller=annotator'.
```

**Comment by nathandunn** [comment]
@childers  That was interesting.  I think this PR will probably fix it.  It very specifically handles the case where addStore is added, as it seems to be the only case where it gets confused.  Very strange. 

Let me know how this works either as a patch or via the PR in #1216.

**Comment by childers** [comment]
I'm happy that I can still find edge cases for you. ( =

In this case, this function is key to how most users can navigate to a particular region, so it is a highly specific use, but not rare at all.

I'm building the patched version now.

**Comment by nathandunn** [comment]
Testing with:

http://localhost:8080/apollo/4/jbrowse/index.html?loc=Group1.25%3A90237..449272&addStores=%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.server%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8796ff4bad7b4dd6b15237abc9bd0cc3%2FTCALIF_genome_v1.0_new_ids.fasta.gff%22%7D%7D&addTracks=%5B%7B%22label%22%3A%22BLAST%2B%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&tracks=DNA%2CAnnotations&highlight=

**Comment by nathandunn** [comment]
Added comment from https://github.com/GMOD/Apollo/issues/1156#issuecomment-241124229 from @childers:

Testing this with the 2.0.4 release patched with pr #1216 

When I click on Show Annotator Panel, it still redirects me to a different scaffold, though it does transmit the added track and proper organism.

<img width="1365" alt="apollo-redirect-before" src="https://cloud.githubusercontent.com/assets/421248/17823048/9fabb93e-6628-11e6-8553-82b59a04ba57.png">
<img width="1365" alt="apollo-redirect-after" src="https://cloud.githubusercontent.com/assets/421248/17823050/9fb224a4-6628-11e6-85b3-18eab1fe3887.png">

**Comment by nathandunn** [comment]
@childers  To clarify, I am not seeing what you are [from above](https://github.com/GMOD/Apollo/issues/1214#issuecomment-241158149) so I want to make sure I'm doing the same thing:

1 - login
2 - view a track in full-screen mode with the "addStoreTracks"
3 - click on "Show Annotator Panel"

Expected: View the same scaffold in the same place with the addStore track preserved.
Observed: Shows a different scaffold.

**Comment by childers** [comment]
@nathandunn I'm sorry but I got confused somewhere in the thread.  
From what I understand:

1. You were able to replicate the original issue in this thread, and issued a patch to correct the bad redirect. 
2. Your patch fixed the URI issue.
3. In my testing, switching between full screen and paneled views does not pass the scaffold.
4. In your testing, switching between full screen and paneled views does not pass the scaffold.

I just cleared my cache and opened a private browser window and the behavior in point 3. is still happening.  

The original issue of the bad URI is fixed with this patch in pr #1216

**Comment by nathandunn** [comment]
For 4, it does fix it (the way I tested it).  However, I may have missed a use-case.    [This comment](https://github.com/GMOD/Apollo/issues/1214#issuecomment-241158548) is just confirming what you see.

**Comment by nathandunn** [comment]
Failure mode is:
1 - jbrowse, with the addStoreTracks, not logged in
2 - click on login within here
3 - click on "show annotator panel" 

Organism is preserved, but the scaffold in 1 and 2 is not shown in 3.

**Comment by nathandunn** [comment]
@childers  I tested this with your branch and it seemed to work for me, with the exception that I get a 404 error on the BLAST track because I don't have access to that server.   Works with the blast track either selected or not.    I used https://github.com/childers/Apollo/tree/my-2.0.4, (```git clone https://github.com/childers/Apollo/ ; cd Apollo ; git checkout my-2.0.4 ; git pull ;```)

Anyway, I tried:

given: a denovo database, and a single admin user:

0 - add the honeybee genome and click on it to make it public
1 - I logout- 
2 - I click on browse genomes and select honeybee
3 - I append this (should only be the one organism so should work), which should match the URL you had in step 2

```http://localhost:8080/apollo/1/jbrowse/index.html?loc=Group1.25%3A90237..449272&addStores=%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.server%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8796ff4bad7b4dd6b15237abc9bd0cc3%2FTCALIF_genome_v1.0_new_ids.fasta.gff%22%7D%7D&addTracks=%5B%7B%22label%22%3A%22BLAST%2B%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&tracks=DNA%2CAnnotations&highlight=```

Case 1: Click on "AnnotatorPanel"
Action: Takes you to a login screen.  Enter credentials.  
Actual==Expected: Takes you back to the Annotator  Panel with the BLAST track is available. 

Case 2: Click on "Login" (when logged out again).  For some reason I have to do this a few times, but it works eventually.  
Action: Has the login popup. Enter credentials. 
Actual==Expected: Puts you in full screen mode and the BLAST track is available.

**Comment by nathandunn** [comment]
From talking with @childers  and @mpoelchau . . . testing with two users they were being switched to the others scaffold.

**Comment by nathandunn** [comment]
I'm repeating [this test](https://github.com/GMOD/Apollo/issues/1214#issuecomment-242258628) with two users going to separate scaffolds.

**Comment by nathandunn** [comment]
Doing this with two users I only see the correct response (the original scaffold they were on).

**Comment by mpoelchau** [comment]
@nathandunn: I'm not surprised! When I copied your workflow (https://github.com/GMOD/Apollo/issues/1214#issuecomment-242258628), I didn't get the error, either. We do get the error consistently on our setup, though. I haven't had enough time to figure out what the difference between your test and our setup is.

**Comment by nathandunn** [comment]
@mpoelchau / @childers  I think the problem is that if both of you copy in "&clientToken=ABC" into the URL then the server things you are the same browser and treats you as such.    If you remove the clientToken, it will automatically generate a unique one for your browser and you shouldn't run into it anymore.

**Comment by nathandunn** [comment]
@childers  I tested replace model as well.    

You are missing a closing ')' in [SequenceTrack.js](https://github.com/childers/Apollo/blob/my-2.0.4/client/apollo/js/View/Track/SequenceTrack.js#L1039) (error shows up in the Developmemnt Console in Chrome).  

Other than that, it actually works accept for the tag is always "replace", but that might be what you wanted.

**Comment by childers** [comment]
@nathandunn Thank you! I can't believe I missed that.

My goal for this release is to have a tag that is always 'replace'. Once we can create custom attribute fields I'll be happy to not change the attribute field.

**Comment by nathandunn** [comment]
@childers  I pushed a fix in #1216 that should handle this case.  It was using the organism name or ID to set the client_token . . . which of course would be shared if you were looking at the same organism.

**Comment by nathandunn** [comment]
@childers  Were you able to confirm if this is working?

**Comment by nathandunn** [comment]
Current error is that when going from a JBrowse screen when logged out:

```
/apollo/<species-name>/jbrowse/?loc=Scaffold1186:145061..145979&addStores={%22url%22:{%22type%22:%22JBrowse/Store/SeqFeature/GFF3%22,%22urlTemplate%22:%22https://gmod-dev.nal.usda.gov/webapp/media/blast/task/8a3be3bab6fa45eb8b7a2e2b298daf41/Cscul.scaffolds.50_new_ids.fa.gff%22}}&addTracks=[{%22label%22:%22BLAST+%20Results%22,%22category%22:%220.%20Reference%20Assembly%22,%22type%22:%22WebApollo/View/Track/DraggableBLASTFeatures%22,%22store%22:%22url%22,%22style%22:{%22renderClassName%22:%22gray-center-10pct%22,%22subfeatureClasses%22:{%22match_part%22:%22blast-match_part%22}}}]&tracks=BLAST+%20Results
```



- [ ] Logging in from the JBrowse screen does a redirect to a client token, but  redirects to the wrong scaffold when logged in:

```
/apollo/1934428378761394903775631985/jbrowse/index.html?addTracks=%5B%7B%22label%22%3A%22BLAST%20%20Results%22%2C%22category%22%3A%220.%20Reference%20Assembly%22%2C%22type%22%3A%22WebApollo%2FView%2FTrack%2FDraggableBLASTFeatures%22%2C%22store%22%3A%22url%22%2C%22style%22%3A%7B%22renderClassName%22%3A%22gray-center-10pct%22%2C%22subfeatureClasses%22%3A%7B%22match_part%22%3A%22blast-match_part%22%7D%7D%7D%5D&addStores%3D%7B%22url%22%3A%7B%22type%22%3A%22JBrowse%2FStore%2FSeqFeature%2FGFF3%22%2C%22urlTemplate%22%3A%22https%3A%2F%2Fgmod-dev.nal.usda.gov%2Fwebapp%2Fmedia%2Fblast%2Ftask%2F8a3be3bab6fa45eb8b7a2e2b298daf41%2FCscul.scaffolds.50_new_ids.fa.gff%22%7D%7D=&loc%3DScaffold1186%3A145061..145979=&tracks%3DBLAST%2B%2BResults=&loc=Scaffold2043%3A1..165198&tracks=DNA%2CAnnotations%2CBLAST%20%20Results&highlight=
```

- clicking on the AnnotatorPanel also goes to the wrong scaffold:

```
/apollo/annotator/index?clientToken=409571437972529684491&addStores={%22url%22:{%22type%22:%22JBrowse/Store/SeqFeature/GFF3%22,%22urlTemplate%22:%22https://gmod-dev.nal.usda.gov/webapp/media/blast/task/8a3be3bab6fa45eb8b7a2e2b298daf41/Cscul.scaffolds.50_new_ids.fa.gff%22}}=&addStores={%22url%22:{%22type%22:%22JBrowse/Store/SeqFeature/GFF3%22,%22urlTemplate%22:%22https://gmod-dev=[nal.usda.gov/webapp/media/blast/task/8a3be3bab6fa45eb8b7a2e2b298daf41/Cscul.scaffolds.50_new_ids.fa.gff%22}}:,%20nal:[usda.gov/webapp/media/blast/task/8a3be3bab6fa45eb8b7a2e2b298daf41/Cscul.scaffolds.50_new_ids.fa.gff%22}}:,%20usda:[gov/webapp/media/blast/task/8a3be3bab6fa45eb8b7a2e2b298daf41/Cscul.scaffolds.50_new_ids.fa.gff%22}}:,%20gov/webapp/media/blast/task/8a3be3bab6fa45eb8b7a2e2b298daf41/Cscul:[scaffolds.50_new_ids.fa.gff%22}}:,%20scaffolds:[50_new_ids.fa.gff%22}}:,%2050_new_ids:[fa.gff%22}}:,%20fa:[gff%22}}:]]]]]]]&tracks=BLAST++Results=&addTracks=[{%22label%22:%22BLAST%20%20Results%22,%22category%22:%220.%20Reference%20Assembly%22,%22type%22:%22WebApollo/View/Track/DraggableBLASTFeatures%22,%22store%22:%22url%22,%22style%22:{%22renderClassName%22:%22gray-center-10pct%22,%22subfeatureClasses%22:{%22match_part%22:%22blast-match_part%22}}}]&tracks=DNA,Annotations,BLAST%20%20Results&highlight=&loc=Scaffold1186:145061..145979=&loc=Scaffold1186:145061=[.145979:,%20:[145979:]]
```

**Comment by nathandunn** [comment]
@childers  I'll try to get to this sometime this week have been stuck on other things.  I am running into these as well.  2 steps forward and 1 step back.

**Comment by nathandunn** [comment]
- [ ] remove URL encoding (which was tripping of the mapping) in favor of manually building the base URL.

**Comment by nathandunn** [comment]
- [ ] add geb test?

Condition 1:
- jbrowse-mode logged out

Condition 2:
- jbrowse-mode logged in

Condition 3:
- jbrowse-mode logged out with "addStore"

Condition 4:
- jbrowse-mode logged in with "addStore"

Action 1:
- click on valid link
- login
- annotator panel

Action 2:
- click on valid link
- annotator panel (and then login)

**Comment by monicacecilia** [comment]
Additional observations from @rwieck 

_"While using the links of the format_ 

_[URL]/Apollo/anopheles_stephensi/jbrowse/index.html?loc=KB664843:2583561..2629699&tracks=gene_model_

_we can't actually drag anything into the User-created Annotations as we were able to [do] before.  We can go into the annotator view, then back (which changes the url to include the clientToken parameter) which then works, but we obviously can't hard code in clientTokens for our links._

_We have tried using the organism id rather than the name, but that does not work either."_

**Comment by nathandunn** [comment]
I reproduced this on icebox and locally, which is running 2.0.4.  Going to test locally against master to see if this is fixed.  There was an issue where it was setting the organism as the client_token, stored in the preferences.     I have a fix on #1216 that fixes this, but there are potentially other issues to fix.  I think that removing the encoding fixes and attempting to fix the weird encoding issue semi-manually will be the way to go.
